nipype/algorithms/confounds.py

```diff
@@ -597,29 +647,37 @@ def _run_interface(self, runtime):
                 header='\t'.join(header),
                 comments='')
 
+        save_metadata = self.inputs.save_metadata
+        if save_metadata:
+            if isinstance(save_metadata, bool):
+                metadata_file = os.path.abspath('component_metadata.tsv')
+            else:
+                metadata_file = save_metadata
+            components_names = np.empty(len(metadata['mask']),
+                dtype='object_')
+            retained = np.where(metadata['retained'])
```

Comment:
I think we would save a lot of trouble using pandas here. Maybe @effigies or @satra correct me, but I'm a bit scared of all this massaging here.